uniref-rest/src/main/java/org/uniprot/api/uniref/controller/UniRefEntryController.java

```diff
@@ -97,18 +96,15 @@ public UniRefEntryController(
                             flags = {Pattern.Flag.CASE_INSENSITIVE},
                             message = "{search.invalid.id.value}")
                     String id,
-            @ModelFieldMeta(
-                            reader = ReturnFieldMetaReaderImpl.class,
-                            path = "uniref-return-fields.json")
-                    @ValidReturnFields(uniProtDataType = UniProtDataType.UNIREF)
-                    @Parameter(
-                            description =
-                                    "Comma separated list of fields to be returned in response")
-                    @RequestParam(value = "fields", required = false)
-                    String fields,
-            HttpServletRequest request) {
-        UniRefEntry entry = entryService.findByUniqueId(id);
-        return super.getEntityResponse(entry, fields, request);
+            @Valid @ModelAttribute UniRefIdRequest idRequest,
+            HttpServletRequest request,
+            HttpServletResponse response) {
+        UniRefEntryResult entryResult = entryService.getEntity(id, idRequest);
+        if (entryResult.getPage() != null) {
+            this.eventPublisher.publishEvent(
```

Comment:
Optional comment, Objects.nonNull can be used to avoid !=null

Reply:
done